server/src/game/role/recruiter.rs

```diff
@@ -64,12 +64,29 @@ impl RoleStateImpl for Recruiter {
             Priority::Kill => {
                 let actor_visits = actor_ref.untagged_night_visits_cloned(game);
                 if let Some(visit) = actor_visits.first(){
-                    if Recruiter::night_ability(self.clone(), game, actor_ref, visit.target) {
-                        if choose_attack {
-                            actor_ref.set_role_state(game, Recruiter{recruits_remaining: self.recruits_remaining.saturating_add(1), ..self})
-                        }else{
-                            actor_ref.set_role_state(game, Recruiter{recruits_remaining: self.recruits_remaining.saturating_sub(1), ..self});
+                    let target_ref = visit.target;
+
+                    let choose_attack = if let Some(IntegerSelection(x)) = game.saved_controllers.get_controller_current_selection_integer(
+                        ControllerID::role(actor_ref, Role::Recruiter, 1)
+                    ){x==0}else{true};
+            
+                    if choose_attack {
+                        target_ref.try_night_kill_single_attacker(
+                            actor_ref,
+                            game,
+                            GraveKiller::RoleSet(RoleSet::Mafia),
+                            AttackPower::Basic,
+                            false
+                        );
+                    } else {
+                        if AttackPower::Basic.can_pierce(target_ref.defense(game)){
+                            MafiaRecruits::recruit(game, target_ref);
+                        } else {
+                            actor_ref.push_night_message(game, ChatMessageVariant::YourConvertFailed);
                         }
+                        actor_ref.set_role_state(game, RoleState::Recruiter(Recruiter{
+                            recruits_remaining: self.recruits_remaining.saturating_sub(1)
+                        }));
```

Comment:
Still lose the recruit charge even if convert fails? Making sure this is on purpose